### PR TITLE
feat: update spring-boot to 3.x

### PR DESCRIPTION
**Pull request by LukasLohoff**
## Description

Updates spring boot to 3.1.x / spring to 6.1 and also updates related dependencies:

- `spring` 5.x -> 6.1 
- `hibernate` 5.x -> 6.2.7
- `hibernate-types-52` -> `hypersistence-utils-hibernate-62` (see https://github.com/vladmihalcea/hypersistence-utils#migrating-to-version-3)
- `springdoc-openapi` 1.6.14 -> 2.2.0
- `keycloak-admin-client` 19.0.3 -> 22.0.1
- `git-commit-id-plugin` -> 6.0.0
- Apache `HttpClient` in HttpUtil (4 -> 5)
- minor improvements / cleanup

@terrestris/devs Please review

### Migration Guide

- update java EE 8 dependencies to jakarta EE 9 (see https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide#jakarta-ee)
- update repositories
  - update `@QueryHints` annotations
  - e.g. `org.hibernate.annotations.QueryHints.CACHEABLE` -> `AvailableHints.HINT_CACHEABLE`
  - if you use custom data fetching methods, migration steps for hibernate 6 / 6.1 / 6.2 might be necessary
    - especially consider https://github.com/hibernate/hibernate-orm/blob/6.0/migration-guide.adoc#removals
    - for more information see the hibernate migration guides listed below
- update your SecurityConfigs which extends `KeycloakWebSecurityConfig` or `SimpleWebSecurityConfig`
   - update your filter chain according to the the spring security migration guides below
   - replace antMatchers with requestMatchers, ignoringAntMatchers with ignoringRequestMatchers
   - update your rules for swagger (`/v3/api-docs`) if they're not already updated
- update your `git-commit-id-plugin` configuration and check if the version set in shogun is overridden (for more information see  https://github.com/terrestris/shogun/pull/730)

### Useful resources

- https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide
- https://docs.spring.io/spring-security/reference/migration/index.html
- https://docs.spring.io/spring-security/reference/5.8/migration/index.html
- https://springdoc.org/v2/
- https://github.com/hibernate/hibernate-orm/blob/6.0/migration-guide.adoc
- https://github.com/hibernate/hibernate-orm/blob/6.1/migration-guide.adoc
- https://github.com/hibernate/hibernate-orm/blob/6.2/migration-guide.adoc
- https://thorben-janssen.com/migrating-to-hibernate-6/

Todos:

- [ ] migration guide
- [x] fix circular spring references
- [ ] check why / if we need hibernate dependencies
- [x] `jakarta-mail.version` is not found
- [x] check if replacement for `SpringPhysicalNamingStrategy` is needed
- [x] fix tests

## Pull request type

- [ ] Bugfix
- [ ] Feature
- [x] Dependency updates
- [ ] Tests
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

## Do you introduce a breaking change?

- [x] Yes
- [ ] No

## Checklist

- [x] I understand and agree that the changes in this PR will be licensed under the 
  [Apache Licence Version 2.0](https://github.com/terrestris/shogun/blob/main/LICENSE).
- [x] I have followed the [guidelines for contributing](https://github.com/terrestris/shogun/blob/main/CONTRIBUTING.md).
- [x] The proposed change fits to the content of the [code of conduct](https://github.com/terrestris/shogun/blob/main/CODE_OF_CONDUCT.md).
- [ ] I have added or updated tests and documentation, and the test suite passes (run `mvn test` locally).
- [ ] I have added a screenshot/screencast to illustrate the visual output of my update.

### CODE REVIEW

**Review by dnlkoch — APPROVED** [review]
Awesome @LukasLohoff!

File: pom.xml [hunk 1]
```diff
@@ -461,6 +517,11 @@
       </dependency>
 
       <!-- Jackson -->
+      <dependency>
+        <groupId>com.fasterxml.jackson.module</groupId>
+        <artifactId>jackson-module-jakarta-xmlbind-annotations</artifactId>
+        <version>2.13.2</version>
```
Comment (ahennr) [inline pom.xml:523]: Minor: Please move version to properties section

File: shogun-boot/src/test/resources/jdbc.properties [hunk 2]
```diff
@@ -15,7 +15,7 @@
 #  limitations under the License.
 shogun.model.packages=de.terrestris.shogun.lib.model
 
-hibernate.dialect=org.hibernate.spatial.dialect.postgis.PostgisPG95Dialect
+hibernate.dialect=org.hibernate.dialect.PostgreSQLDialect
```
Comment (ahennr) [inline shogun-boot/src/test/resources/jdbc.properties:18]: spatial queries still work?

**Review by ahennr — APPROVED** [review]
Nive work @LukasLohoff

File: shogun-boot/src/test/resources/jdbc.properties (hunk 2 above)
[the remark this answers is not in the document]
Comment (LukasLohoff) [reply shogun-boot/src/test/resources/jdbc.properties:18]: in Hibernate 6 the automatic dialect resolution was improved and specifying the specific dialect and its version shouldn't be required anymore 

see https://vladmihalcea.com/hibernate-dialect/ for example

File: pom.xml [hunk 3]
```diff
@@ -102,24 +102,33 @@
...
-    <spring-boot.version>2.7.6</spring-boot.version>
-    <spring-data.version>2.7.6</spring-data.version>
+    <spring-boot.version>3.1.3</spring-boot.version>
+    <spring-data.version>3.1.3</spring-data.version>
+
+    <!-- Mail -->
+    <jakarta-mail.version>2.0.1</jakarta-mail.version>
+
+    <!-- Hibernate -->
+    <hibernate.version>6.2.7.Final</hibernate.version>
```
[the remark this answers is not in the document]
Comment (LukasLohoff) [reply pom.xml]: yes, now with spring boot 3.1.3 the versions match. Fixed in 84f61cd6c832b8ee5a58e65b6e7cb45daf4e21c4

File: pom.xml (hunk 1 above)
Comment (LukasLohoff) [reply pom.xml]: fixed in d368194853e5d80c261c1881cd29d45a05c56d2d

File: shogun-boot/src/test/resources/jdbc.properties (hunk 2 above)
Comment (LukasLohoff) [reply shogun-boot/src/test/resources/jdbc.properties:18]: yes!

**Pull request merged**